src/bsoncxx/document/value.cpp

```diff
@@ -23,6 +23,9 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace document {
 
+value::value() noexcept : _data(NULL, [](std::uint8_t*) {}) {
```

Comment:
I'm just as happy with nullptr.  Pointers have a legitimate "not there" sentinel, no need to add an optional on top.

I kind of prefer making the uint deleter null check, or having a noop deleter to the lambda though